dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/muzzle/ReferenceMatcher.java

```diff
@@ -183,15 +184,31 @@ public boolean matches(final ClassLoader loader) {
     return mismatches;
   }
 
+  private static boolean matchesPrimitive(String longName, String shortName) {
+    // The two meta type systems in use here differ in their treatment of primitive type names....
```

Comment:
Which two type systems?

Reply:
One is from ByteBuddy and the other is from the otel auto agent.  Their APIs are so similar and this code is so annoying because of it.